embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/EmbracePerformanceInfoService.kt

```diff
@@ -69,10 +88,23 @@
         logDeveloper("EmbracePerformanceInfoService", "Building performance info")
 
         return PerformanceInfo(
-            diskUsage = metadataService.getDiskUsage()?.copy(),
-            memoryWarnings = memoryService.getCapturedData()?.toList(),
-            networkInterfaceIntervals = networkConnectivityService.getCapturedData()?.toList(),
-            powerSaveModeIntervals = powerSaveModeService.getCapturedData()?.toList(),
+            diskUsage = captureSafely { metadataService.getDiskUsage()?.copy() },
+            memoryWarnings = captureSafely { memoryService.getCapturedData()?.toList() },
+            networkInterfaceIntervals = captureSafely {
+                networkConnectivityService.getCapturedData()?.toList()
+            },
+            powerSaveModeIntervals = captureSafely {
+                powerSaveModeService.getCapturedData()?.toList()
+            },
         )
     }
+
+    private inline fun <R> captureSafely(result: () -> R): R? {
+        return try {
+            result()
+        } catch (exc: Throwable) {
+            InternalStaticEmbraceLogger.logger.logError("Exception thrown capturing session data", exc)
```

Comment:
Can we pass in the logger instance we initialize in the service so this would bubble up while we run integration tests?